tools/common/Assembly.cs

```diff
@@ -328,6 +328,15 @@ public static LinkWithAttribute GetLinkWithAttribute (CustomAttribute attr)
 			return linkWith;
 		}
 
+		void AddFramework (string file)
+		{
+			Framework framework;
+			if (Driver.GetFrameworks (App).TryGetValue (file, out framework) && framework.Version > App.SdkVersion)
+				Driver.Log (3, "Not linking with the framework {0} (referenced by a module reference in {1}) because it was introduced in {2} {3}, and we're using the {2} {4} SDK.", file, FileName, App.PlatformName, framework.Version, App.SdkVersion);
```

Comment:
I know you're just moving code around but this one should be a warning (not just a log) since the code can fail at runtime (e.g. works locally, build on a bot with different Xcode, packaged product fails)

Reply:
Agree, but that would be everyone build with an older Xcode would have that warning. I'm assuming that is ok.